### PR TITLE
Update the documentation of the provenance interface

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Update the documentation of the provenance interface so it is in line with the ESMValTool documentation here: https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance

Link to documentation: https://esmvaltool--1305.org.readthedocs.build/projects/ESMValCore/en/1305/interfaces.html#information-provided-by-the-diagnostic-script-to-esmvalcore

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: doc/interfaces.rst [hunk 1]
```diff
@@ -105,15 +105,14 @@ Information provided by the diagnostic script to ESMValCore
 
 After the diagnostic script has finished running, ESMValCore will try to store provenance information. In order to
 link the produced files to input data, the diagnostic script needs to store a YAML file called :code:`diagnostic_provenance.yml`
-in it's :code:`run_dir`.
+in its :code:`run_dir`.
 
-For output file produced by the diagnostic script, there should be an entry in the :code:`diagnostic_provenance.yml` file.
-The name of each entry should be the path to the output file.
-Each file entry should at least contain the following items
+For every output file produced by the diagnostic script, there should be an entry in the :code:`diagnostic_provenance.yml` file.
```
Comment (schlunma) [inline doc/interfaces.rst:110]:
```suggestion
For every output file (netCDF files, plot files, etc.) produced by the diagnostic script, there should be an entry in the :code:`diagnostic_provenance.yml` file.
```

File: doc/interfaces.rst [hunk 2]
```diff
@@ -105,15 +105,14 @@ Information provided by the diagnostic script to ESMValCore
...
+The name of each entry should be the path to the file.
+Each output file entry should at least contain the following items
 
-- :code:`ancestors` a list of input files used to create the plot
+- :code:`ancestors` a list of input files used to create the plot.
 - :code:`caption` a caption text for the plot
```
Comment (schlunma) [inline doc/interfaces.rst:115]:
```suggestion
- :code:`caption` a caption text for the plot.
```

File: doc/interfaces.rst [hunk 3]
```diff
@@ -123,31 +122,30 @@ An example :code:`diagnostic_provenance.yml` file could look like this
...
     domains: [global]
-    plot_file: /path/to/recipe_output/plots/diagnostic_name/script_name/CMIP5_MPI-ESM-LR_Amon_historical_r1i1p1_T2Ms_pr_2000-2002_mean.png
-    plot_type: zonal
+    plot_types: ['zonal']
     references: [acknow_project]
     statistics: [mean]
 
 You can check whether your diagnostic script successfully provided the provenance information to the ESMValCore by
-verifying that
+verifying by checking the following points:
```
Comment (schlunma) [inline doc/interfaces.rst:144]:
```suggestion
checking the following points:
```

**Comment by schlunma** [comment]
Can you merge the latest `main` into this branch to let the checks pass?

File: doc/interfaces.rst [hunk 4]
```diff
@@ -106,15 +106,14 @@ Information provided by the diagnostic script to ESMValCore
...
 link the produced files to input data, the diagnostic script needs to store a YAML file called :code:`diagnostic_provenance.yml`
-in it's :code:`run_dir`.
+in its :code:`run_dir`.
 
-For output file produced by the diagnostic script, there should be an entry in the :code:`diagnostic_provenance.yml` file.
-The name of each entry should be the path to the output file.
-Each file entry should at least contain the following items
+For every output file (netCDF files, plot files, etc.) produced by the diagnostic script, there should be an entry in the :code:`diagnostic_provenance.yml` file.
+The name of each entry should be the path to the file.
+Each output file entry should at least contain the following items
```
Comment (schlunma) [inline doc/interfaces.rst:113]:
```suggestion
Each output file entry should at least contain the following items:
```

**Review by schlunma — COMMENTED** [review]
I cannot see the bullet points in the rendered output, but apart from that it looks fine!

**Comment by bouweandela** [comment]
I also noticed the problem with the bullet points: https://github.com/ESMValGroup/ESMValCore/issues/1179#issuecomment-863001132, I don't think it's related to this pull request.

**Pull request merged**